[XuHh03] iP

XuHh03 opened the pull request.

XuHh03 committed:
Level 0: Rename, Greet, Exit

XuHh03 committed:
added echo functionality and exit command for Nova chatbot

XuHh03 committed:
added list feature and display on request to Nova chatbot

XuHh03 committed:
added new task class and add mark done feature to Nova chatbot

XuHh03 committed:
modified code according to coding standards

XuHh03 committed:
added support for Todos, Deadlines, and Events using inheritance

V1T0bh commented on src/main/java/Nova.java:
```diff
@@ -29,11 +29,28 @@ public static void main(String[] args) {
+                Task t = tasks[taskNumber - 1];
+                t.markAsNotDone();
+                System.out.println("____________________________________________________________");
+                System.out.println(" OK, I've maked this task as not done yet:");
```
typo found
"maked" -> "marked"

V1T0bh commented on src/main/java/Nova.java:
```diff
@@ -4,6 +4,9 @@ public class Nova {
+
         System.out.println("""
                 ____________________________________________________________
                 Hello! I'm Nova
```
I like your chatbot's name!

V1T0bh commented on src/main/java/Task.java:
```diff
@@ -5,8 +5,8 @@ public class Task {
     public Task(String description) {
```
constructor is properly declared. Good job.

V1T0bh commented on src/main/java/Nova.java:
```diff
@@ -48,6 +48,26 @@ public static void main(String[] args) {
+                String taskDescription = userInput.substring(5);
+                Task newTask = new Todo(taskDescription);
+                tasks[taskCount++] = newTask;
+                printTaskAdded(newTask, taskCount);
```
lines repeated in multiple if statements. overall code should be shortened if possible.

V1T0bh commented on src/main/java/Task.java:
```diff
@@ -21,7 +21,7 @@ public void markAsNotDone() {
     @Override
     public String toString() {
-        return "[" + getStatusIcon() + "]" + description;
+        return "[" + getStatusIcon() + "] " + description;
```
good catch

V1T0bh reviewed (commented):
Overall code looks good and complete. Way ahead of my progress. Keep up the good work!

Sheeeesh-code commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,90 @@
+        while (true) {
+            String userInput = scanner.nextLine();
+
+            if (userInput.equals("bye")) {
```
Your if, else if, else if is well written

Sheeeesh-code commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+public class Event extends Task{
+    protected String from;
```
Good job, the name of your variable are explicit

Sheeeesh-code commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,27 @@
+        this.isDone = true;
+    }
+
+    public void markAsNotDone() {
```
Good Job, your methods name and class name are explicits

Sheeeesh-code commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,90 @@
+                System.out.println("____________________________________________________________");
+            } else if (userInput.startsWith("unmark")) {
+                int taskNumber = Integer.parseInt(userInput.split(" ")[1]);
+                Task t = tasks[taskNumber - 1];
```
I think the name of the variable task could be more explicit

Sheeeesh-code commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,90 @@
+
+public class Nova {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
```
Good job not to many global variable but don't forget to close the scanner

XuHh03 committed:
added test for textUI

added tests including todo, deadline, event, list, bye

XuHh03 committed:
added abstractions and improve readability

XuHh03 committed:
updated design for Nova

improved welcome message
changed lineSeparator

XuHh03 committed:
updated error throwing logic

XuHh03 committed:
Merge branch 'branch-Level-5' into master

XuHh03 committed:
modified error handling with exception

XuHh03 committed:
added packages and changed directory in testing

packages: ui, task, exception
updated README.md

XuHh03 committed:
Merge branch 'branch-A-Packages' into master

LuoYu-uwu commented on src/main/java/Nova/ui/Nova.java:
```diff
@@ -0,0 +1,182 @@
+        printLineSeparator();
+    }
+
+    private static void handleMark(String userInput, boolean done) throws NovaException {
```
Perhaps rename `done` to sound more like a boolean

LuoYu-uwu commented on src/main/java/Nova/task/Todo.java:
```diff
@@ -0,0 +1,12 @@
+package Nova.task;
```
Perhaps rename this as package name should be all lowercase

LuoYu-uwu commented on src/main/java/Nova/ui/Nova.java:
```diff
@@ -0,0 +1,182 @@
+import java.util.Scanner;
+
+public class Nova {
+    private static final int maxTaskCount = 100;
```
Perhaps rename this as constant names should be in all uppercase using underscore to separate words

LuoYu-uwu reviewed (commented):
The code is generally clear and readable, with well-structured classes, just take note of some naming conventions. good job.

XuHh03 committed:
modified to collection for tasks

XuHh03 committed:
added delete method

updated test to include delete test

XuHh03 committed:
added local save logic to Nova

saves tasks to Nova.txt

XuHh03 committed:
Resolved merge conflict

XuHh03 committed:
Created Jar file for Nova

updated local storage logic for different OS

XuHh03 committed:
Refactor code to improve OOP design

- extracted TextUi class to handle user interactions
- extracted Parser class to parse user inputs
- extracted TaskList class to manage task list operations
- updated Nova.java to delegate responsibilities to different classes
- created Main.java file to handle program startup

XuHh03 committed:
updated test cases to incorprate OOP refactor

XuHh03 committed:
updated gitignore to ignore Data file

XuHh03 committed:
add LocalDateTime parsing and formatting to tasks

- Event and Deadline accepts input in d/m/yyyy format
- Ensures consistent saving and loading in Nova.txt
- Displays dates in better readability

XuHh03 committed:
added find command to search tasks Nova

XuHh03 committed:
Merge pull request #1 from XuHh03/branch-level-9

added find command to search tasks Nova

XuHh03 committed:
Added JavaDoc comments to classes and methods in Nova

XuHh03 committed:
Merge pull request #2 from XuHh03/branch-A-JavaDoc

Added JavaDoc comments to classes and methods in Nova

XuHh03 committed:
added user guide for Nova

XuHh03 committed:
changed images to code for better readability

XuHh03 committed:
fixed bug in Github Page
